astropy/io/fits/time.py

```diff
@@ -0,0 +1,174 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+from collections import OrderedDict
+
+import numpy as np
+
+from . import Header, Card
+
+from ...coordinates import EarthLocation
+from ...table import Table, QTable, Column
+from ...time import Time
+from ...time.formats import FITS_DEPRECATED_SCALES
+from ...units import Quantity
+
+
+class FitsTime(object):
+    """
+    A class to maintain the metadata of time columns stored in a FITS binary table
+    (header and data) and to convert those time columns into `~astropy.time.Time` columns.
+
+    Parameters
+    ----------
+    nfields : int
+        Number of fields(columns) in the binary table.
+    """
+
+    def __init__(self, nfields):
+        """
+        Set default time metadata.
+        """
+        # Default time scale UTC
+        self.scale = 'UTC'
+        # If none of the reference values are provided, MJD = 0.0 must be assumed.
+        self.mjd = None
+        # Default time unit
+        self.unit = 's'
+        self.time_columns = OrderedDict()
+        for i in range(1, nfields + 1):
+            self.time_columns[i] = None
+
+    def set_global_time(self, key, value, comment):
+        """
+        Set the global time reference frame attributes.
+
+        Parameters
+        ----------
+        key : string
+            FITS global time reference frame keyword.
+        value : int, str, list
+            value associated with specified keyword.
+        comment : str
+            comment associated with specified keyword.
+        """
+        if key == 'TIMESYS':
+            self.scale = value
+        elif key == 'MJDREF':
+            self.mjd = value
+        elif key == 'JDREF':
+            self.jd = value
+        elif key == 'DATEREF':
+            self.date = value
+        elif key == 'TREFPOS':
+            self.pos = value
+        elif key == 'TREFDIR':
+            self.dir = value
+        elif key == 'TIMEUNIT':
+            self.unit = value
+        elif key == 'TIMEOFFS':
+            self.offs = value
+        elif key == 'OBSGEO-X':
+            self.loc_x = value
+        elif key == 'OBSGEO-Y':
+            self.loc_y = value
+        elif key == 'OBSGEO-Z':
+            self.loc_z = value
+
+    def set_column_override(self, key, value, comment):
+        """
+        Set the time column specific override attributes.
+
+        Parameters
+        ----------
+        key : string
+            FITS time column specific keyword.
+        value : int, str, list
+            value associated with specified keyword.
+        comment : str
+            comment associated with specified keyword.
+        """
+        idx = int(key[-1])
```

Comment:
Then use a mapping to avoid the repetition ("Don't Repeat Yourself"):
```
base_map = {'TCTYP': 'scale', 'TRPOS': 'pos', ..}
if base in base_map:
    self.time_columns[idx][base_map[base]] = value
else:
    # Raise ValueError since this probably should not ever happen.